src/orion/core/io/resolve_config.py

```diff
@@ -127,6 +127,9 @@ def fetch_default_options():
         for _, key, default_value in env_vars:
             default_config[signifier][key] = default_value
 
+    if 'serverSelectionTimeoutMS' not in default_config['database']:
+        default_config['database']['serverSelectionTimeoutMS'] = 5000
```

Comment:
I would put this instead in the default values of the init of MongoDB database since this option is specific to this backend.